autoload/go/test.vim

```diff
@@ -122,6 +122,31 @@ function! go#test#Func(bang, ...) abort
   call call('go#test#Test', args)
 endfunction
 
+" Testcase runs a single subtest of a function for a subtest name on the
+" current cursor line.
+" Arguments are passed to the `go test` command.
+function! go#test#Case(bang, ...) abort
+  let l:testParent = go#util#TestName()
+
+  " Check for a struct initialization on the line, and take its value
+  let l:searchPattern = '\s*\:\s*"\zs\w*\ze'
```

Comment:
Good call about the regex there, you're right it is a little sloppy. I don't need to use the preceding whitespace for the back anchor, I'll chop that out. I'll also some testing to cover that the pattern works as expected for names with and w/o whitespace. It looks like `go test -run` will take a subtest name and transform the spaces for me. So I'm happy to let it do that instead of more string manipulation here.